### PR TITLE
Add new method `addCollectibleVerifyOwnership`

### PR DESCRIPTION
**Pull request by adonesky1**
It seemed to not make much sense to make a call to the CollectiblesController in the background to validate ownership, immediately before calling the same controller to add that collectible. This PR adds a new method `addCollectibleVerifyOwnership` which wraps the addCollectible method together with the validation step.

### CODE REVIEW

File: src/assets/CollectiblesController.test.ts [hunk 1]
```diff
@@ -181,387 +187,491 @@ describe('CollectiblesController', () => {
...
-    expect(collectiblesController.state.collectibleContracts).toHaveLength(0);
   });
 
-  it('should not remove collectible contract if collectible still exists', async () => {
-    await collectiblesController.addCollectible('0x01', '1', {
-      name: 'name',
-      image: 'image',
-      description: 'description',
-      standard: 'standard',
+  describe('addCollectibleVerifyOwnership', () => {
```
Comment (adonesky1) [inline src/assets/CollectiblesController.test.ts:534]: TLDR: I reorganized the tests such that each major method has its own describe block.

File: src/assets/CollectiblesController.test.ts [hunk 2]
```diff
@@ -181,387 +187,491 @@ describe('CollectiblesController', () => {
...
+        tokenId: '1234',
+      });
     });
 
-    await collectiblesController.addCollectible('0x01', '2', {
-      name: 'name',
-      image: 'image',
-      description: 'description',
-      standard: 'standard',
+    it('should throw an error if selected address is not owner of input collectible', async () => {
```
Comment (adonesky1) [inline src/assets/CollectiblesController.test.ts:561]: This is the only new test in this file. The rest are just shuffled around.

**Review by gantunesr — DISMISSED** [review]
LGTM. Just a little nit, the method name, but I have no better suggestion

File: src/assets/CollectiblesController.test.ts [hunk 3]
```diff
@@ -181,387 +187,491 @@ describe('CollectiblesController', () => {
...
+        collectiblesController.state.collectibleContracts[0],
+      ).toStrictEqual({
+        address: ERC721_KUDOSADDRESS,
+        name: 'KudosToken',
+        symbol: 'KDO',
+      });
     });
-  });
 
-  it('should add collectible by selected address', async () => {
```
Comment (Gudahtt) [inline src/assets/CollectiblesController.test.ts:377]: It looks like this test is no longer preset. Was it removed accidentally?

I see that there is now an equivalent test for your new method, but not for `addCollectible`, which is what this tested

**Review by wachunei — DISMISSED** [review]

File: src/assets/CollectiblesController.test.ts (hunk 3 above)
Comment (adonesky1) [reply src/assets/CollectiblesController.test.ts:377]: Yes I replaced it with a test for my method. I will add it back for `addCollectible` and rename it slightly and keep it for testing the new method as well? Sound good @Gudahtt?

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**